pkg/venafi/tpp/connector_test.go

```diff
@@ -528,6 +530,110 @@ func TestRequestCertificateWithValidHours(t *testing.T) {
 	DoRequestCertificateWithValidHours(t, tpp)
 }
 
+// The reason we are using a mock HTTP server rather than the live TPP server is
+// because consistently triggering the 500 error in a stage different than 0
+// requires putting a powershell script on the TPP VM, which is not something
+// that can be done as part of the "set up" of the current tests we have in
+// vcert.
+func TestRequestCertificateWithReset(t *testing.T) {
+	tests := []struct {
+		name             string
+		givenResetStatus int
+		givenResetBody   string
+		expectRequest    bool
+		expectErr        string
+	}{
+		{
+			name:             "should request certificate when reset returns 200",
+			givenResetStatus: 200,
+			expectRequest:    true,
+		},
+		{
+			name:             "should request certificate when the certificate DN is not found",
+			givenResetStatus: 400,
+			givenResetBody:   `{"Error": "CertificateDN error. CertificateDN: \"\\VED\\Policy\\Foo\\test.com\" does not exist or you do not have sufficient rights to the object."}`,
+			expectRequest:    true,
+		},
+		{
+			name:             "should request certificate when no previous enrollment exists",
+			givenResetStatus: 400,
+			givenResetBody:   `{"Error": "Reset is not completed. No reset is required for the certificate."}`,
+			expectRequest:    true,
+		},
+		{
+			name:             "should not request a certificate when reset returns an unexpected 401",
+			givenResetStatus: 401,
+			givenResetBody:   `{"error":"session_error","error_description":"Grant rejected scope 'certificate:manage'.}`,
+			expectRequest:    false,
+			expectErr:        "while resetting certificate in case of previous enrollment: 401 Unauthorized, body: ''",
```

Comment:
Note: I don't know why the body is `''`. I am expecting the body to be `{"error":"session_error","error_description":"Grant rejected scope 'certificate:manage'.}`, but it seems `request()` doesn't work like that...

Reply:
Ah, that was my fault. I fixed that in aad355f3d712fd870b5b29ead57441d2e949da60.